### DISCUSSION TITLE
Create a new language transpiler

### DISCUSSION THREAD

**Discussion by dtrendafilov** [SofiaCPP/IPL#113] [Projects]
Create a language that gets transpiled to another high-level programming language or another VM with high level IR.

Possible targets:
- JavaScript
- C++
- [NekoVM](https://nekovm.org)
- WebAssembly
- LLVM IR

The source and target languages must be different enough to make sense for the translation (i.e. stripping type annotations from TypeScript to get JavaScript will be considered too easy)

**Comment by ghost** [comment]
Може ли да ползваме LLVM IR като таргет за този проект?
 
И къде се избира проект - тук в коментарите или има някакво друго място?

**Comment by herince** [comment]
Ако е подходящо за проект, както @alexvitkov попита, искам да направя transpiler от C до LLVM IR. Като може би по-скоро на някакъв subset на C (най-вероятно просто без пойнтри, защото не знам колко сложно ще е да има и pointers).

**Reply by dtrendafilov** [reply]
LLVM IR is a bit  low-level, ще се получи компилатор, но е ок.

**Reply by ghost** [reply]
В такъв случай, ако е ок, бих искал и аз като проект да напиша компилатор от някакъв мой си C-like език -> LLVM IR.

**Reply by NDobrev** [reply]
Ok

**Comment by GTSimeonov** [comment]
Искам за проект да напиша C до Javascript транспилатор

**Reply by NDobrev** [reply]
OK

**Comment by dtagarev** [comment]
Здравейте искам да правя този проект
Мисля да сведа lua до с++ или някакъв intermidiate language.

**Reply by dtrendafilov** [reply]
Супер. Каква част от lua? Какви типове от стандартната библиотека ще се поддържат?
Предлагам да са поне всичко от езика и само числа от типовете.

**Comment by Syndamia** [comment]
Здравейте, искам да направя проект от Common Lisp до LLVM IR. Със сигурност няма да успея да сведа всичко, но поне повечето важни компоненти ще направя.

**Reply by dtrendafilov** [reply]
Нека да дефинираме кои са тези компоненти?
